packages/sdk/src/FuelWalletConnection.ts

```diff
@@ -16,16 +16,59 @@ import type { JSONRPCRequest } from 'json-rpc-2.0';
 
 import { WindowConnection } from './connections/WindowConnection';
 import { getTransactionSigner } from './utils/getTransactionSigner';
+import { hasWindow } from './utils/hasWindow';
+
+const queue: JSONRPCRequest[] = [];
 
 export class FuelWalletConnection extends WindowConnection {
+  timeoutTimer: NodeJS.Timer;
+  isListenerAdded = false;
+
   acceptMessage(message: MessageEvent<CommunicationMessage>): boolean {
     const { data: event } = message;
     return (
       message.origin === window.origin && event.target === PAGE_SCRIPT_NAME
     );
   }
 
+  constructor() {
+    super();
+
+    const executeQueue = async () => {
+      let request = queue.pop();
+      while (request) {
+        await this.sendRequest(request);
+        request = queue.pop();
+      }
+
+      clearInterval(this.timeoutTimer);
+    };
+
+    const handleFuelInjected = () => {
+      if (hasWindow) {
+        if (!this.isListenerAdded) {
+          this.attachEventListener();
+          this.isListenerAdded = true;
+        }
+        if (window.fuel) {
+          executeQueue();
+        }
+      }
+    };
+
+    handleFuelInjected();
+
+    this.timeoutTimer = setInterval(handleFuelInjected, 1000);
```

Comment:
We should abstract this in a better way splitting the functions on utilities or on the body of the class.